networkit/cpp/coarsening/GraphCoarsening.cpp

```diff
@@ -14,7 +14,14 @@ GraphCoarsening::GraphCoarsening(const Graph& G) : Algorithm(), G(&G) {
 
 }
 
-Graph GraphCoarsening::getCoarseGraph() const {
+const Graph& GraphCoarsening::getCoarseGraph() const {
+    if(!hasRun) {
+        throw std::runtime_error("Call run()-method first.");
+    }
```

Comment:
Here you can use `assureFinished`

Reply:
It's really the same except for a slightly different message isnt it? 
Shouldnt the other methods be using `assureFinished()` then too?

Reply:
Yes, it's the standard way of assuring that the algorithm was executed. It was probably introduced later than some classes so we still have some instances of `if (!hasRun) throw ...`

Reply:
Alright, i replaced it in all three functions.